File: matlab/collisions/getCentersRails.m
function output = getCentersRails(numOfRails)

% Create Rail Centers
centers1 = complex([0:8:(28*8)],0);
centers2 = centers1(1:end-1) + 8 * exp(1i * pi / 3);
centers = [centers1 centers2].' + 8 * exp(i*2*pi/3);
rails = centers;
for jj = 1:(numOfRails-1)
    rails = [rails; centers + jj * 16 * exp(1i * 2 * pi/3)];
end
output = rails;
end

File: python/ics/cobraOps/getCentersRails.py

import numpy as np


def getCentersRails(numOfRails):
    # Set the number of cobras per rail
    cobrasPerRail = 29 + 28
    
    # Create the centers array
    centers = np.zeros(numOfRails * cobrasPerRail, dtype="complex")
    
    # Fill the first rail
    firstRail = centers[:cobrasPerRail]
    firstRail[:29] = 8 * np.arange(29)
    firstRail[29:] = 8 * np.arange(28) + 8 * np.exp(1j * np.pi / 3)
    firstRail += 8 * np.exp(1j * 2 * np.pi / 3)
    
    # Order the first rail center by the x coordinate
    orderedIndexes = firstRail.sort()
    firstRail[:] = firstRail[orderedIndexes]
    
    # Fill the rest of the rails
    for i in xrange(1, numOfRails):
        centers[i * cobrasPerRail:(i + 1) * cobrasPerRail] = firstRail + i * 16 * np.exp(1j * 2 * np.pi / 3)
    
    return centers
